### ISSUE TITLE
crash under python2.7

### ISSUE THREAD

**Issue by e3kr** [PyCQA/pyflakes#654]
Running pyflakes 2.3.1 on this file

```python
# fail.py 
import unused

def fail(**unused):
    pass
```

results in

```python
$ pyflakes fail.py 
Traceback (most recent call last):
  File "/home/kr/pyflakes/.env/bin/pyflakes", line 8, in <module>
    sys.exit(main())
  File "/home/kr/pyflakes/.env/lib/python2.7/site-packages/pyflakes/api.py", line 210, in main
    warnings = checkRecursive(args, reporter)
  File "/home/kr/pyflakes/.env/lib/python2.7/site-packages/pyflakes/api.py", line 155, in checkRecursive
    warnings += checkPath(sourcePath, reporter)
  File "/home/kr/pyflakes/.env/lib/python2.7/site-packages/pyflakes/api.py", line 99, in checkPath
    return check(codestr, filename, reporter)
  File "/home/kr/pyflakes/.env/lib/python2.7/site-packages/pyflakes/api.py", line 74, in check
    w = checker.Checker(tree, file_tokens=file_tokens, filename=filename)
  File "/home/kr/pyflakes/.env/lib/python2.7/site-packages/pyflakes/checker.py", line 916, in __init__
    self.checkDeadScopes()
  File "/home/kr/pyflakes/.env/lib/python2.7/site-packages/pyflakes/checker.py", line 1041, in checkDeadScopes
    self.report(messg, node, value.name, value.source)
  File "/home/kr/pyflakes/.env/lib/python2.7/site-packages/pyflakes/checker.py", line 1047, in report
    self.messages.append(messageClass(self.filename, *args, **kwargs))
  File "/home/kr/pyflakes/.env/lib/python2.7/site-packages/pyflakes/messages.py", line 32, in __init__
    Message.__init__(self, filename, loc)
  File "/home/kr/pyflakes/.env/lib/python2.7/site-packages/pyflakes/messages.py", line 12, in __init__
    self.lineno = loc.lineno
AttributeError: 'arguments' object has no attribute 'lineno'
```

while it runs properly under python3:

```shell
$ pyflakes fail.py 
fail.py:2:1 'unused' imported but unused
fail.py:4:12 redefinition of unused 'unused' from line 2
```
----

Under 2.7 it also fails on
```python
# fail.py 
import unused

def fail(*unused):
    pass
```
(difference is in `*unused` vs `**unused`)

When the last star is removed it starts working again on python 2.7:
```python
$ cat fail.py 
# fail.py 
import unused

def fail(unused):
    pass
$ pyflakes fail.py 
fail.py:2:1 'unused' imported but unused
fail.py:4:10 redefinition of unused 'unused' from line 2
```

----

`import`-ed symbols are somewhat different, as the following runs silently on both python2 and python3:

```python
# fail.py 
unused = True

def fail(**unused):
    pass
```

**Comment by asottile** [comment]
I think it's probably time we stop supporting python 2

**Comment by e3kr** [comment]
I was afraid of this response :(

**Comment by asottile** [comment]
it's almost 2 years past end of life, it's time to move on

**Comment by hugovk** [comment]
https://github.com/PyCQA/pyflakes/pull/660 has been merged, ending support for Python 2.7.

**Comment by sigmavirus24** [comment]
@e3kr older versions of pyflakes will still support Python 2.7. That language is not evolving and so pyflakes will continue to work, you likely don't need any of the new features of pyflakes that are all related to new language features in newer versions of 3.x
